tiledb/sm/array/array.cc

```diff
@@ -453,9 +474,56 @@ const EncryptionKey& Array::get_encryption_key() const {
 }
 
 Status Array::reopen() {
-  return reopen(utils::time::timestamp_now_ms());
+  std::unique_lock<std::mutex> lck(mtx_);
```

Comment:
Let's re-use the `reopen(uint64_t timestamp)` API here like before, replacing `utils::time::timestamp_now_ms()` with the config value.